### PR TITLE
Use new-style RBAC in flask-sqlalchemy backend

### PR DESCRIPTION
**Pull request by gw000**

### CODE REVIEW

File: backends/flask-sqlalchemy/app/authorization.polar [hunk 1]
```diff
@@ -1,71 +1,100 @@
...
+  permissions = ["read"];
+  relations = { parent: Org };
+  "read" if "list_role_assignments" on "parent";
+}
+
+resource RepoRole {
+  permissions = ["read"];
+  relations = { parent: Repo };
+  "read" if "list_role_assignments" on "parent";
+}
```
Comment (gw000) [inline backends/flask-sqlalchemy/app/authorization.polar:73]: I added these as resources so I could include them in the `checked_permissions` map that `sqlalchemy-oso` uses to decide what resources are visible to a user; without it they'd be invisible. If anyone knows a better way to do this I'll remove them.

File: backends/flask-sqlalchemy-oso/app/authorization.polar [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+has_relation(org: Org, "parent", repo: Repo) if repo.org = org;
+has_relation(repo: Repo, "parent", issue: Issue) if issue.repo = repo;
+
+has_relation(org: Org, "parent", role: OrgRole) if org = role.org;
+has_relation(repo: Repo, "parent", role: RepoRole) if repo = role.repo;
+
+allow(actor, action, resource) if
+  has_permission(actor, action, resource);
+
+# TODO remove
```
Comment (leina05) [inline backends/flask-sqlalchemy-oso/app/authorization.polar:95]: Should be able to remove this now.

File: backends/flask-sqlalchemy-oso/app/authorization.polar [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    role matches { name: name, repo_id: repo.id };
+
+has_relation(org: Org, "parent", repo: Repo) if repo.org = org;
+has_relation(repo: Repo, "parent", issue: Issue) if issue.repo = repo;
+
+has_relation(org: Org, "parent", role: OrgRole) if org = role.org;
+has_relation(repo: Repo, "parent", role: RepoRole) if repo = role.repo;
+
+allow(actor, action, resource) if
+  has_permission(actor, action, resource);
```
Comment (leina05) [inline backends/flask-sqlalchemy-oso/app/authorization.polar:93]: I think we've been putting this at the top of the file conventionally.

File: backends/flask-sqlalchemy-oso/app/authorization.polar [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+type has_permission(_: Actor, _: String, _: Actor);
+
+# Users can see each other.
+has_permission(_: User, "read", _: User);
+
+# A User can read their own profile.
+has_permission(_: User{id: id}, "read_profile", _:User{id: id});
+
+# Any logged-in user can create a new org.
+has_permission(_: User, "create", _: Org);
```
Comment (leina05) [inline backends/flask-sqlalchemy-oso/app/authorization.polar:105]: Potentially reorganize pending standardizing the organization conventions.

**Review by leina05 — APPROVED** [review]
Looks great!

File: backends/flask-sqlalchemy-oso/README.md [hunk 5]
```diff
@@ -9,24 +9,22 @@ authorization.
...
 
 ## Backend
 
 ### Running tests
 
 ```console
-$ cd backends/flask-sqlalchemy
+$ cd backends/flask-sqlalchemy-oso
 $ python3 -m venv venv && source venv/bin/activate
 $ pip3 install -r requirements.txt -r requirements-dev.txt
```
Comment (gj) [inline backends/flask-sqlalchemy-oso/README.md:25]: nit: feel like it'd be nice to wrap these commands up into a Makefile task like `make requirements` in the `backends/flask-sqlalchemy-oso` directory and then have the `test-flask-sqlalchemy-oso` task depend on `requirements` so the entire set of commands you need to run tests is `make -C tests test-flask-sqlalchemy-oso`

File: backends/flask-sqlalchemy-oso/app/__init__.py [hunk 6]
```diff
@@ -127,10 +122,10 @@ def init_oso(app, Session: sessionmaker):
     oso = SQLAlchemyOso(Base)
 
     # Enable roles features.
-    oso.enable_roles(User, Session)
+    #    oso.enable_roles(User, Session)
```
Comment (gj) [inline backends/flask-sqlalchemy-oso/app/__init__.py:125]: kill this & the comment on the line above?

File: backends/flask-sqlalchemy-oso/app/authorization.polar [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+
+has_relation(org: Org, "parent", role: OrgRole) if org = role.org;
+has_relation(repo: Repo, "parent", role: RepoRole) if repo = role.repo;
+
+allow(actor, action, resource) if
+  has_permission(actor, action, resource);
+
+# TODO remove
+type has_permission(_: Actor, _: String, _: Actor);
+
```
Comment (gj) [inline backends/flask-sqlalchemy-oso/app/authorization.polar:97]: can we remove this now that https://github.com/osohq/oso/pull/1158 has gone in?

```suggestion
```

File: backends/flask-sqlalchemy-oso/README.md (hunk 5 above)
Comment (gw000) [reply backends/flask-sqlalchemy-oso/README.md:25]: 👌  lmswicd

File: backends/flask-sqlalchemy-oso/app/authorization.polar (hunk 3 above)
Comment (gw000) [reply backends/flask-sqlalchemy-oso/app/authorization.polar]: moved it

File: backends/flask-sqlalchemy-oso/README.md (hunk 5 above)
Comment (gw000) [reply backends/flask-sqlalchemy-oso/README.md]: Ok, you can now do that (for this backend)

File: backends/flask-sqlalchemy-oso/Makefile [hunk 8]
```diff
@@ -9,18 +9,28 @@ dev: .make.requirements-install
...
+test-server: .make.requirements-test
 	FLASK_APP="app:create_app('sqlite:///test.db', False)" flask run
 
-.make.requirements-install: requirements.txt
+.make.requirements-install: venv-source requirements.txt
 	pip install -r requirements.txt
 	touch $@
 
+venv:
+	python -m venv venv
```
Comment (gj) [inline backends/flask-sqlalchemy-oso/Makefile:20]: maybe?
```suggestion
	python3 -m venv venv
```

**Pull request merged**